src/rbd.cc

```diff
@@ -499,6 +475,7 @@ static int do_clone(librbd::RBD &rbd, librados::IoCtx &p_ioctx,
   if ((features & RBD_FEATURE_LAYERING) != RBD_FEATURE_LAYERING) {
     return -EINVAL;
   }
+  features &= ~RBD_FEATURE_STRIPINGV2;
 
   return rbd.clone(p_ioctx, p_name, p_snapname, c_ioctx, c_name, features,
```

Comment:
Due to an earlier bug, we always used to add the stripingv2 feature bit for clones here. Because of that, we made the kernel client accept it when it was using the default kind of striping (allowing only changing object size). So adding stripingv2 when it's not doing anything special won't cause any harm.